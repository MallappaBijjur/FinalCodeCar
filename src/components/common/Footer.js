import React from 'react';
import './footer.css';

const Footer = () => (
  <footer className='footer'>
    <p className='footer__text' >Â© AUTO1 Group 2018</p>
  </footer>
);

export default Footer;
