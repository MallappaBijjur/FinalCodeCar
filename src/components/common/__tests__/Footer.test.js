import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import rerender from 'react-test-renderer';
import Footer from '../Footer';

describe("Footer", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
	it("renders without crashing", ()=>{
		const div = document.createElement("div");
		ReactDOM.render(<Footer />, div);
	});
	
	it('renders Filter component', () => {
        const { getByText } = render(<Footer />);   
		const linkElement = getByText(/Â© AUTO1 Group 2018/i);
		expect(linkElement).toBeInTheDocument();
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<Footer />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});




