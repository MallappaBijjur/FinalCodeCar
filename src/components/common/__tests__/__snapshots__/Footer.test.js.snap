// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer
  className="footer"
>
  <p
    className="footer__text"
  >
    Â© AUTO1 Group 2018
  </p>
</footer>
`;
